### ISSUE TITLE
Visual cue for updating/checking in CodeDemo: Loading bar

### ISSUE THREAD

**Issue by agoscinski** [agoscinski/scicode-widgets-unofficial#10]
Coming from PR #9 

We can add a loading bar gif to the self._demo_button_box in CodeDemo. Not sure what the best way is. This is one simple way to do
```python
import ipywidgets
with open("/home/alexgo/lib/scicode-widgets/python/scwidgets/loading.gif", "rb") as file:
    # read file as string into `image` 
    image = file.read()

    img = ipywidgets.Image(
        layout=ipywidgets.Layout(width='2%'),
        value=image,
        format='gif')
display(img)

# later in code ...
with open("/home/alexgo/lib/scicode-widgets/python/scwidgets/loading_red.gif", "rb") as file:
    # read file as string into `image` 
    img.set_value_from_file(file)
```
This requires to open a file all the time, we can load the binary into the memory, but not sure if this is a good solution.

**Comment by agoscinski** [comment]
From @jprado1 
> I think it would be useful to give the option to not have visual cues in "automatic update mode", as constant changes in parameters can cause a strange flickering

Yes for fast functions the feedback is weird, I agree it has to be different. The automatic update mode should disable the buttons  without visual cues the loading gif. I think about jumping on initialization directly to UPDATING status and disable the visual cues already during UPDATING while keeping the loading gif.

**Comment by jprado1** [comment]
> I think about jumping on initialization directly to UPDATING status and disable the visual cues already during UPDATING while keeping the loading gif.

I like this idea and think it would resolve most of the flickering issue. After some tests, it seems that another issue is that, when written outputs are before pyplot outputs, clearing and loading the output makes plot jump when the written output is loaded. Nevertheless this is a lesser problem that can be solved by consistently putting written output only after plots (and never before).
